ast/ast.go

```diff
@@ -121,6 +125,20 @@ type (
 	}
 )
 
+// A BinaryExpression is any expression of the form "operand operator operand"
+// including arithmetic (+, -, *, /), logical operators (and, or), and
+// comparisons (=, !=, <, <=, >, >=)
+type BinaryExpression struct {
+	Node
+	Operator string     `json:"operator"`
+	LHS      Expression `json:"lhs"`
+	RHS      Expression `json:"rhs"`
+}
+
+func (*BinaryExpression) exprNode() {}
+func (*Literal) exprNode()          {}
+func (*FieldRead) exprNode()        {}
```

Comment:
Shouldn't FieldCall also get an `exprNode()`?

Reply:
At least for now, FieldCall is part of the filter syntax but not part of the expression syntax.  I think that the easiest short-term thing is to keep the two separate and then, once function calls are part of the expression syntax we can look at either combining related AST nodes if that makes sense or renaming them to clarify their roles if they remain separate...